aggregator/pom.xml

```diff
@@ -55,6 +55,18 @@
             <version>${project.version}</version>
             <classifier>${spark.version.classifier}</classifier>
         </dependency>
+        <!-- set provided scope for JNI and ucx to avoid pulling into aggregator jar  -->
```

Comment:
If we cannot find a way to automatically pull in immediate provided dependencies, there needs to be a comment here explaining that any changes here likely impact the corresponding dependency list maintained in dist/pom.xml.